### PR TITLE
Add support for mds without predefined attention mask

### PR DESCRIPTION
**Pull request by ohallstrom**
**Changes**

This PR enables reading mds data that does not have a predefined attention_mask.

As the attention mask is redundant to store in mds in our standard case (as it is just an array of ones with the same length as the input_ids), we have removed it in the new batches of data that we are processing. This PR accommodates this update, enabling training with this new format.

With this PR, supporting mds-data requiring less storage, the reads from disk is significantly reduced.

Additionally the conversion script of mds-data has been made more efficient.

**Tests**
I have taken mds-data with predefined attention masks, that I then have removed the attention masked from using the conversion script. Using the original data, and the new data without attention masks, I have achieved close to identical loss curves with and without predefined attention masks. I have verified this both with and without streaming.

image
image

### CODE REVIEW

**Comment by NohTow** [comment]
LGTM
